opendevin/runtime/utils/runtime_build.py

```diff
@@ -105,30 +109,30 @@ def _generate_dockerfile(
     dockerfile_content += (
         'RUN if [ -d /opendevin/code ]; then rm -rf /opendevin/code; fi\n'
     )
-    # Unzip the tarball to /opendevin/code
+    # Unzip the tarball to /opendevin/code and install dependencies via poetry
     dockerfile_content += (
         'RUN cd /opendevin && tar -xzvf project.tar.gz && rm project.tar.gz\n'
-    )
-    dockerfile_content += f'RUN mv /opendevin/{source_code_dirname} /opendevin/code\n'
-
-    # ALTERNATIVE, but maybe not complete? (toml error!)
-    dockerfile_content += (
-        'RUN cd /opendevin/code && '
-        '/opendevin/miniforge3/bin/mamba run -n base poetry env use python3.11 && '
-        '/opendevin/miniforge3/bin/mamba run -n base poetry install --no-interaction --no-root\n'
-        'RUN /opendevin/miniforge3/bin/mamba run -n base poetry cache clear --all . && \\\n'
-        'apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* &&\\\n'
-        '/opendevin/miniforge3/bin/mamba clean --all\n'
+        f'RUN mv /opendevin/{source_code_dirname} /opendevin/code\n'
+        'WORKDIR /opendevin/code\n'
+        'RUN /opendevin/miniforge3/bin/mamba run -n base poetry env use python3.11\n'
+        # Activate the opendevin virtual environment
+        'RUN source /opendevin/code/.venv/bin/activate\n'
```

Comment:
If you store .venv here under `/opendevin/code`, every time you want to update the project directory, we will have to re-install everything from scratch, instead of updating existing dependency. Because at line 110, you will `rm -rf /opendevin/code` if it exists before updating code. So i previously suggest we just leave out the whole chum from L80-L87 for simplicity.